MdeModulePkg/Library/VarCheckHiiLib/VarCheckHiiLibStandaloneMM.c

```diff
@@ -0,0 +1,135 @@
+/** @file
+
+  Implementation functions and structures for var check services.
+  This file provides functions and structures to register and handle variable checks
+  in the Standalone MM environment, specifically for HII variables.
+
+Copyright (c) 2024, Intel Corporation. All rights reserved.<BR>
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/MmServicesTableLib.h>
+#include <Protocol/MmCommunication.h>
+#include <Library/VarCheckLib.h>
+
+#include "VarCheckHii.h"
+#include "VarCheckHiiLibCommon.h"
+
+extern EFI_GUID                gEfiVariableCheckHiiCommunicationGuid;
+VAR_CHECK_HII_VARIABLE_HEADER  *mVarCheckHiiBinMmRecieved    = NULL;
+UINTN                          mVarCheckHiiBinMmSizeRecieved = 0;
+
+/**
+  Registers a handler for HII variable checks in MM environment.
+  This function is intended to be called to register a handler for checking variables
+  in the Standalone MM environment. It allocates memory for the variable
+  check data and copies the data from the communication buffer.
+
+  @param[in] DispatchHandle    The handle of the dispatch function.
+  @param[in] Context           Optional context for the handler, not used in this implementation.
+  @param CommBuffer        The buffer of data being passed in.
+  @param CommBufferSize    The size of the data being passed in.
+  @retval EFI_SUCCESS           Registration and memory allocation were successful.
+  @retval EFI_INVALID_PARAMETER The CommBuffer or CommBufferSize is NULL.
+  @retval EFI_ACCESS_DENIED     The buffer size is invalid or the buffer is in an invalid location.
+  @retval EFI_OUT_OF_RESOURCES  Memory allocation for the variable check data failed.
+
+**/
+EFI_STATUS
+EFIAPI
+VarCheckHiiLibRegisterMmHandler (
+  IN EFI_HANDLE  DispatchHandle,

```

Comment:
VarCheckHiiLibRegisterMmHandler->VarCheckHiiLibReceiveHiiBinHandler.
The corresponding handler GUID can be gVarCheckReceiveHiiBinHandlerGuid.